src/cli/commands/force/lightning/lwc/preview.ts

```diff
@@ -87,53 +89,103 @@ export default class Preview extends Setup {
         `$ sfdx force:lightning:lwc:preview -p Android -t LWCEmu2 -n HellowWordComponent`
     ];
 
+    // NOTE: The following properties are just place holders to help with typescript compile.
+    protected title: string = '';
+    protected fulfilledMessage: string = '';
+    protected unfulfilledMessage: string = '';
+
+    private serverPort: string = '';
+    private deviceName: string = '';
+    private componentName: string = '';
+    private targetApp: string = '';
+    private projectDir: string = '';
+    private configFilePath: string = '';
+    private appConfig:
+        | IOSAppPreviewConfig
+        | AndroidAppPreviewConfig
+        | undefined;
+
     public async run(): Promise<any> {
-        const platform = this.flags.platform;
-        this.logger.info(`Preview command invoked for ${platform}`);
-        return Promise.all<void, SetupTestResult>([
-            this.validateAdditionalInputs(),
-            super.run()
-        ])
+        const logger = this.logger;
+        logger.info(`Preview command invoked for ${this.flags.platform}`);
+
+        return this.validateAdditionalInputs() // first validate input parameters
             .then((result) => {
-                this.logger.info(
-                    'Setup requirements met, continuing with preview'
-                );
+                // then validate setup requirements
+                if (
+                    PreviewUtils.isTargetingBrowser(this.targetApp) ||
+                    (this.appConfig &&
+                        this.appConfig.preview_server_enabled === true)
+                ) {
+                    const extraReqs: Requirement[] = [
+                        {
+                            checkFunction: this.isLwcServerRunning,
+                            fulfilledMessage: messages.getMessage(
+                                'reqs:server:fulfilledMessage'
+                            ),
+                            logger,
+                            title: messages.getMessage('reqs:server:title'),
+                            unfulfilledMessage: messages.getMessage(
+                                'reqs:server:unfulfilledMessage'
+                            )
+                        }
+                    ];
+                    super.addRequirements(extraReqs);
```

Comment:
Nit: seems to defeat the overridable nature of `addRequirements()` to call it as `super` here. Any reason we wouldn't simpy use `this` instead?

Reply:
yes it should be `this` not `super` 🤦‍♂️